src/js/angular/graphexplore/controllers/rdf-class-hierarchy.controller.js

```diff
@@ -488,6 +488,14 @@ function RdfClassHierarchyCtlr($scope, $rootScope, $location, $repositories, $li
     };
 
     $scope.isAllGraphsSelected = function () {
-        return $scope.getSelGraphValue() === 'all.graphs.label'
-    }
+        return $scope.getSelGraphValue() === 'all.graphs.label';
+    };
+
+    //Method used to limit the graphs loaded in the drop-down menu, otherwise browsers crash.
+    $scope.downsizeGraphsInRepoIfMoreThan1000 = function () {
```

Comment:
You don't need this function in both rdf-class-hierarchy and dependencies controllers. You could add this check and slice when you're assigning the returned value from RDF4JRepositoriesRestService.resolveGraphs() in initView function of the controllers.